src/frequenz/sdk/timeseries/_resampling.py

```diff
@@ -334,6 +480,94 @@ def add_sample(self, sample: Sample) -> None:
             sample: The sample to be added to the buffer.
         """
         self._buffer.append(sample)
+        if self._source_properties.sampling_start is None:
+            self._source_properties.sampling_start = sample.timestamp
+        self._source_properties.received_samples += 1
+
+    def _update_source_sample_period(self, now: datetime) -> bool:
+        """Update the source sample period.
+
+        Args:
+            now: The datetime in which this update happens.
+
+        Returns:
+            Whether the source sample period was changed (was really updated).
+        """
+        assert (
+            self._buffer.maxlen is not None and self._buffer.maxlen > 0
+        ), "We need a maxlen of at least 1 to update the sample period"
+
+        config = self._config
+        props = self._source_properties
+
+        # We only update it if we didn't before and we have enough data
+        if (
+            props.sampling_period_s is not None
+            or props.sampling_start is None
+            or props.received_samples
+            < config.resampling_period_s * config.max_data_age_in_periods
+            or len(self._buffer) < self._buffer.maxlen
+        ):
```

Comment:
should this also check `props.received_samples` is equal to zero? or is it redundant?

Reply:
it is redundant as `max_data_age_in_periods` is ensured to be greater than 1 and `resampling_period_s` is checked for positive